pages/docs/concepts/asyncapi-document/extending-specification.md

```diff
@@ -0,0 +1,103 @@
+
+The benefits of extending the AsyncAPI specification include enhanced specificity — delivering a clear encoding of necessary API details to clients, and consistency — allowing reusable components across the API. However, extensions should be used sparingly. Since they go beyond the Specification, they may not be universally understood.
+
+## AsyncAPI Specification Extensions
+
+The AsyncAPI Specification allows the addition of custom properties through patterned fields prefixed with `x-`. This feature enables the accommodation of specific functionalities not initially included in the AsyncAPI specification.
+
+The `x-` prefix is used to define custom properties. These properties are user-defined and won't conflict with future specification versions because any property starting with `x-` is reserved for user definitions, and the AsyncAPI organization will not use this prefix for any of their future standard properties.
+
+```mermaid
+flowchart TB
+    subgraph "AsyncAPI Specification"
+        A[Specification] --> B[Custom Properties]
+    end
+    subgraph "Custom Properties"
+        B --> C[Patterned fields prefixed with x-]
+        B --> D[Reserved for user definitions]
+        B --> E[Support any valid JSON format value]
+    end
+    subgraph "Usage"
+        C --> F[Accommodate specific functionalities]
+        D --> G[Avoid conflict with future versions]
+        E --> H[May or may not be supported by tools]
+    end
+    subgraph "Tooling"
+        H --> I[Internal tools]
+        H --> J[Open-sourced tools]
+        I --> K[Extended support may be added]
+        J --> L[Extended support may be added]
+    end
+```
+
+Here is a simple example of how to extend the AsyncAPI specification:
+
+```yml
+asyncapi: 3.0.0
+info:
+  title: Cool Example
+  version: 0.1.0
+channels:
+  userSignedUp:
+    address: user/signedup
+    messages:
+      userSignedUp:
+        description: An event describing that a user just signed up.
+        $ref: '#/components/messages/UserSignedUp'
+      x-custom-property: Custom Value        
+```
+
+In the above example, under the `user/signedup` channel, a custom property `x-custom-property` is added. The value assigned to this property is `Custom Value`.
+
+The custom properties can have any valid JSON format value. For example, it could be null, a primitive, an array, or an object. The newly added custom property is compatible with the extended tooling and can supplement it with additional functionalities.
+
+<Remember>
+All available tooling might not support AsyncAPI extensions. The tooling can be extended to understand and handle the added data, especially if the tools are internal or open source.
+</Remember>
```

Comment:
>The newly added custom property is compatible with the extended tooling and can supplement it with additional functionalities.

What is `extended tooling`? 🤔 

> All available tooling might not support AsyncAPI extensions

All tools will allow you to use extensions, otherwise, they are not compatible with the AsyncAPI standard, however, the extension will be ignored in most cases (except documentation tools such as react components, HTML template, and hence studio will also display them)

As soon as you use extensions and want to use them for something, you have to expect to write custom tools or as you say here, extend existing tools if they have a feature for this.

Reply:
Removed this part.